conformance/lang/expressions/is-expr/is_expr.balt

```diff
@@ -0,0 +1,1931 @@
+    io:println(x4.attrNon is error); // @output true
+    io:println(x4?.attrNon is ()); // @output true
+}
+
+Test-Case: output
+Description: Test xml attribute access expression as expression in type test expression.
+Fail-Issue: ballerina-platform/ballerina-lang#35190
+Labels: is-expr, nil-type, string, xml-optional-attribute-access-expr, xml:Element
+
+function init() {
+    xml:Element x3 = xml `<elem xmlns="ns-uri" attr="val" xml:space="default"></elem>`;
+    io:println(x3?.attr is string); // @output true
+    io:println(x3?.attrNon is ()); // @output true
+}
+
+Test-Case: output
+Description: Test annotation access expression as expression in type test expression.
+Fail-Issue: ballerina-platform/ballerina-lang#33243
+Labels: annot-access-expr, annotation-decl, any, array-type, explicit-new-expr, is-expr, mapping-constructor-expr,
+        module-class-defn, module-type-defn, record-type, typedesc-type, typeof-expr
+
+type Annot record {
+    string foo;
+    int bar?;
+};
+
+public annotation Annot v1 on type, class;
+
+annotation Annot[] v2 on class;
+
+public annotation Annot v3 on function;
+
+@v1 {
+    foo: "strValue",
+    bar: 1
+}
+type T1 record {
+    string name;
+};
+
+@v1 {
+    foo: "strValue"
+}
+@v2 {
+    foo: "v2 value 1"
+}
+@v2 {
+    foo: "v2 value 2"
+}
+class T2 {
+    string name = "ballerina";
+}
+
+function init() {
+    T1 a = {name: "John"};
+    typedesc<any> t = typeof a;
+    io:println(t.@v1 is Annot); // @output true
+
+    T1 b = {name: "John"};
+    t = typeof b;
+    io:println(t.@v2 is ()); // @output true
+
+    T2 c = new;
+    t = typeof c;
+    io:println(t.@v1 is Annot); // @output true
+    io:println(t.@v2 is Annot[]); // @output true
+    io:println(t.@v3 is ()); // @output true
+}
+
+Test-Case: output
+Description: Test member access expression as expression in type test expression.
+Labels: boolean, float, int, is-expr, mapping-constructor-expr, member-access-expr, module-type-defn, nil-type,
+        optional-type, record-type, string, union-type
+
+type Type string|xml|boolean;
+
+type Record1 record {|
+    int|string a;
+    float? b;
+    int c?;
+    boolean d = true;
+    Type? e;
+    string...;
+|};
+
+function init() {
+    Record1 rec1 = {a: 1, b: (), "e": "A", "f": "B"};
+
+    io:println(rec1["a"] is int); // @output true
+    io:println(rec1["b"]is float); // @output false
+    io:println(rec1["c"] is int); // @output false
+    io:println(rec1["d"] is Type); // @output true
+    io:println(rec1["e"] is string); // @output true
+    io:println(rec1["f"] is string); // @output true
+    io:println(rec1["g"] is ()); // @output true
+}
+
+Test-Case: output
+Description: Test function call expression as expression in type test expression.
+Labels: function-call-expr, int, is-expr, module-const-decl, module-type-defn, nil-type, optional-type, string,
+        union-type
+
+const nm = "Anne Smith";
+
+type Type string;
+
+function init() {
```

Comment:
When possible try to break it down to small tests with different labels

Reply:
The tests were separated for different expressions and types. For example, this particular test mainly focuses on testing the behavior with `function-call-expr` as the expression. Can we break them further?